examples/analytics/airflow-on-eks/main.tf

```diff
@@ -0,0 +1,542 @@
+resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
+  bucket = aws_s3_bucket.this.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
+    }
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "this" {
+  bucket = aws_s3_bucket.this.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  restrict_public_buckets = true
+  ignore_public_acls      = true
+}
+
+#---------------------------------------------------------------
+# Apache Airflow Postgres Metastore DB Master password
+#---------------------------------------------------------------
+resource "random_password" "postgress" {
+  length           = 16
+  special          = true
+  override_special = "![]{}"
+}
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "postgres" {
+  name                    = "postgres"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "postgres" {
+  secret_id     = aws_secretsmanager_secret.postgres.id
+  secret_string = random_password.postgress.result
+}
+
+data "aws_secretsmanager_secret_version" "postgres" {
+  secret_id = aws_secretsmanager_secret.postgres.id
+
+  depends_on = [aws_secretsmanager_secret_version.postgres]
+}
+
+#---------------------------------------------------------------
+# Apache Airflow Webserver Secret
+#---------------------------------------------------------------
+resource "random_password" "airflow_webserver" {
+  length           = 16
+  special          = true
+  override_special = "![]{}"
+}
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "airflow_webserver" {
+  name                    = "airflow_webserver"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id     = aws_secretsmanager_secret.airflow_webserver.id
+  secret_string = random_password.airflow_webserver.result
+}
+
+data "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id = aws_secretsmanager_secret.airflow_webserver.id
+
+  depends_on = [aws_secretsmanager_secret_version.airflow_webserver]
+}
+
+#---------------------------------------------------------------
+# Webserver Secret Key
+#---------------------------------------------------------------
+resource "kubectl_manifest" "airflow_webserver" {
+  sensitive_fields = [
+    "data.webserver-secret-key"
+  ]
+
+  yaml_body = <<YAML
+apiVersion: v1
+kind: Secret
+metadata:
+   name: ${local.airflow_webserver_secret_name}
+   namespace: ${module.airflow_irsa.namespace}
+type: Opaque
+data:
+  webserver-secret-key: ${base64encode(data.aws_secretsmanager_secret_version.airflow_webserver.secret_string)}
+YAML
+
+  depends_on = [module.eks_blueprints.eks_cluster_id]
```

Comment:
because this relies on `module.airflow_irsa.namespace` which relies on the cluster ID, OIDC details, etc., you should be safe to remove the depends here and things will be ordered correctly

Reply:
done